### PR TITLE
std: use `stream_position` where applicable

### PR DESCRIPTION
**Pull request by marcospb19**
by replacing `seek(SeekFrom::Current(0))` calls

### CODE REVIEW

File: library/std/src/fs.rs [hunk 1]
```diff
@@ -984,7 +984,7 @@ impl OpenOptions {
     /// If a file is opened with both read and append access, beware that after
     /// opening, and after every write, the position for reading may be set at the
     /// end of the file. So, before writing, save the current position (using
-    /// <code>[seek]\([SeekFrom]::[Current]\(0))</code>), and restore it before the next read.
+    /// <code>[Seek]::[stream_position]</code>), and restore it before the next read.
```
Comment (marcospb19) [inline library/std/src/fs.rs:987]: This was the motivating piece of documentation, but then I started applying to other places (where it was OK to do so).

**Comment by joshtriplett** [comment]
r=me once CI passes.

**Comment by joshtriplett** [comment]
r? libs

**Comment by JohnCSimon** [comment]
> r=me once CI passes.

From triage - seems like this hasn't been set to review state
@rustbot ready

**Pull request merged**